src/ASM/LR/GlobalNodes.C

```diff
@@ -0,0 +1,187 @@
+        case 5: edge = LR::WEST | LR::SOUTH | LR::TOP; break;
+        case 6: edge = LR::EAST | LR::SOUTH | LR::TOP; break;
+        case 7: edge = LR::WEST | LR::NORTH | LR::TOP; break;
+        case 8: edge = LR::EAST | LR::NORTH | LR::TOP; break;
+      }
+    }
+  } else if (dim == 1) {
+    if (lr.nVariate() == 2) {
+      switch (lidx) {
+        case 1: edge = LR::WEST; break;
+        case 2: edge = LR::EAST; break;
+        case 3: edge = LR::SOUTH; break;
+        case 4: edge = LR::NORTH; break;
+        default: break;
+      }
+    } else {
+      switch (lidx) {
+        case 1: edge = LR::BOTTOM | LR::SOUTH; break;
+        case 2: edge = LR::BOTTOM | LR::NORTH; break;
+        case 3: edge = LR::TOP | LR::SOUTH; break;
+        case 4: edge = LR::TOP | LR::NORTH; break;
+        case 5: edge = LR::BOTTOM | LR::WEST; break;
+        case 6: edge = LR::BOTTOM | LR::EAST; break;
+        case 7: edge = LR::TOP | LR::WEST; break;
+        case 8: edge = LR::TOP | LR::WEST; break;
+        case 9: edge = LR::SOUTH | LR::WEST; break;
+        case 10: edge = LR::SOUTH | LR::EAST; break;
+        case 11: edge = LR::NORTH | LR::WEST; break;
+        case 12: edge = LR::NORTH | LR::EAST; break;
+      }
+    }
+  } else if (dim == 2) {
+    switch (lidx) {
+      case 1: edge = LR::WEST; break;
+      case 2: edge = LR::EAST; break;
+      case 3: edge = LR::SOUTH; break;
+      case 4: edge = LR::NORTH; break;
+      case 5: edge = LR::BOTTOM; break;
+      case 6: edge = LR::TOP; break;
+    }
+  }
+
+  std::vector<LR::Basisfunction*> edgeFunctions;
+  lr.getEdgeFunctions(edgeFunctions, edge);
+
+  if (dim == 1) {
+    if (lr.nVariate() == 2) {
+      int v = (lidx == 1 || lidx == 2) ? 0 : 1;
+      int u = 1-v;
+      ASMLRSpline::Sort(u, v, orient, edgeFunctions);
+    } else {
+      int dir = (lidx-1)/4;
+      int u = dir == 0;
+      int v = 1 + (dir != 2);
+      ASMLRSpline::Sort(u, v, orient, edgeFunctions);
+    }
+  } else if (dim == 2) {
+    int dir = (lidx-1)/2;
+    int u = dir == 0;
+    int v = 1 + (dir != 2);
+    ASMLRSpline::Sort(u, v, orient, edgeFunctions);
+  }
+
+  GlobalNodes::IntVec lNodes;
+  lNodes.reserve(edgeFunctions.size());
+  for (const LR::Basisfunction* func : edgeFunctions)
+    lNodes.push_back(func->getId());
+
+  return lNodes;
+}
+
+
+std::vector<GlobalNodes::IntVec>
+GlobalNodes::calcGlobalNodes (const GlobalNodes::LRSplineVec& pchs,
+                              const GlobalNodes::InterfaceVec& interfaces)
+{
+  // count total number of nodes
+  size_t nNodes = 0;
+  std::vector<GlobalNodes::IntVec> result(pchs.size());
+  auto it = result.begin();
+  for (const auto& pch : pchs) {
+    it->resize(pch->nBasisFunctions());
+    std::iota(it->begin(), it->end(), nNodes);
+    nNodes += pch->nBasisFunctions();
+    ++it;
+  }
+
+  // remap common nodes
+  InterfaceOrder ifOrder;
+  std::set<ASM::Interface, InterfaceOrder> ifset(ifOrder);
```

Comment:
Is it necessary with the `ifOrder` instance here? Thought the following would work fine:

    std::set<ASM::Interface,InterfaceOrder> ifset;

We have a similar construct for the `MPC` sets.
Even better, could the `InterfaceOrder` class be replaced by a local lambda here?

Reply:
removed the instance. cannot use a lambda as a tempalte parameter.